src/main/java/com/synopsys/integration/alert/provider/blackduck/collector/util/BlackDuckResponseCache.java

```diff
@@ -54,20 +58,20 @@ public BlackDuckResponseCache(BlackDuckBucketService blackDuckBucketService, Bla
     }
 
     public <T extends BlackDuckResponse> Optional<T> getItem(Class<T> responseClass, String url) {
-        if (null == responseClass || StringUtils.isBlank(url)) {
+        if (null == responseClass || StringUtils.isBlank(url) || null == blackDuckBucketService) {
```

Comment:
Why would the bucket service be nullable?

Reply:
If you look at the changes for the DistributionFilterFactory and BlackDuckProjectNameExtractor, we were constructing a BlackDuckService on the fly and using that. Now I am creating the BlackDuckBucketService once to be shared during a "batch" of notification processing so that we can cache the requests and responses rather than make the same requests repeatedly.

Reply:
But if you have an invalid Black Duck config, then the BlackDuckBucketService will be null

Reply:
How can we get this far if that is nullable though? Why would we be instantiating a network cache if the config is invalid?

Reply:
I updated the BlackDuckProperties methods so that we use Optional when appropriate and throw exceptions when we should. So  this null check is no longer necessary.